nixpkgs-pin.nix

```diff
@@ -1,10 +1,14 @@
-pkgs:
-
 let
-  # tested nixpgks-unstable version
-  nixpkgsGH = with builtins; (fromJSON (readFile ./flake.lock)).nodes.nixpkgs.locked;
+  defaultNix = (import
+    (
+      let lock = builtins.fromJSON (builtins.readFile ./flake.lock); in
+      builtins.fetchGit {
```

Comment:
Found the following trick here:
https://github.com/ZilchOS/bootstrap-from-tcc/blob/de68f791590dbc48c7f3e738edd086fbe95e7b4b/default.nix#L52

Define a minimal `fetchurl`
```
  fetchurl = { url, sha256 }: derivation {
    name = builtins.baseNameOf url;
    inherit url;
    urls = [ url ];
    unpack = false;

    builder = "builtin:fetchurl";
    system = "builtin";
    outputHashMode = "flat"; outputHashAlgo = "sha256";
    preferLocalBuild = true;
    outputHash = sha256;
  };
```

Reply:
Hm, unfortunately this does not work as it results in a hash mismatch. I guess this is due to `unpack = false`. However, `unpack = true` seems to only work with NARs.